File: app/src/main/java/com/example/ikastyle/UI/GearPowerImageView.java
package com.example.ikastyle.UI;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;

import androidx.appcompat.widget.AppCompatImageView;

import com.example.ikastyle.Common.Const.GearPowerResourceId;
import com.example.ikastyle.R;

/*
 * ギアパワー画像用のImageView
 */
public class GearPowerImageView extends AppCompatImageView implements View.OnTouchListener {
    private int gearPowerKind;

    public GearPowerImageView(Context context, AttributeSet attrs){
        super(context, attrs);

        //カスタム属性gearPowerKindの値を取得しフィールド変数gearPowerKindに入れる
        TypedArray typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.GearPowerImageView, 0, 0);
        gearPowerKind = typedArray.getInt(R.styleable.GearPowerImageView_gearPowerKind, gearPowerKind);

        // onTouchリスナーをセット
        this.setOnTouchListener(this::onTouch);
    }

    public int getGearPowerKind(){
        return gearPowerKind;
    }

    @Override
    public boolean onTouch(View view, MotionEvent motionEvent){
        if(motionEvent.getAction() == MotionEvent.ACTION_DOWN){
            view.startDragAndDrop(null, new View.DragShadowBuilder(view), view, 0);
        }

        return true;
    }
}

File: app/src/main/java/com/splatool/ikastyle/view/customizedView/GearPowerImageView.kt
package com.splatool.ikastyle.view.customizedView

import android.content.Context
import androidx.appcompat.widget.AppCompatImageView
import com.splatool.ikastyle.R
import android.view.View.OnTouchListener
import android.util.AttributeSet
import android.view.*

/*
 * ギアパワー画像用のImageView
 */
class GearPowerImageView(context: Context, attrs: AttributeSet?) :
    AppCompatImageView(context, attrs), OnTouchListener {
    var gearPowerKind: Int = 0

    override fun onTouch(view: View?, motionEvent: MotionEvent?): Boolean {
        if (motionEvent?.action == MotionEvent.ACTION_DOWN) {
            view?.startDragAndDrop(null, DragShadowBuilder(view), view, 0)
        }
        return true
    }

    init {

        //カスタム属性gearPowerKindの値を取得しフィールド変数gearPowerKindに入れる
        val typedArray =
            getContext().obtainStyledAttributes(attrs, R.styleable.GearPowerImageView, 0, 0)
        gearPowerKind = typedArray.getInt(R.styleable.GearPowerImageView_gearPowerKind, gearPowerKind)

        // onTouchリスナーをセット
        setOnTouchListener(this)
    }
}